Unexpected errors in alternative paths for observation Blood Pressure

# Test Case/s To Reproduce Issue
Alternative flows Observation Blood Pressure

# path to test case
tests/robot/OBSERVATION/01_create.robot
# robot command to execute related test case(s) in your terminal/console

robot -d results/ -L TRACE -i blood-pressureAND194 robot

# by tag

061 Create Observation Blood Pressure loinc instant
062 Create Observation Blood Pressure loinc period
063 Create Observation Blood Pressure loinc timing
064 Create Observation Blood Pressure snomed datetime
065 Create Observation Blood Pressure snomed instant
066 Create Observation Blood Pressure snomed period
067 Create Observation Blood Pressure snomed timing



## Actual Result

500 for testcase 004

422 for all other cases


Cases 002, 004, 005, 006, 007 and 008 have no stack trace. 
Robot error log:


case 002:

{'body': {'issue': [{'code': 'processing',
                     'diagnostics': 'The type of element effectiveInstant is not known, which is illegal. Valid types at this point are dateTime, Period',
                     'location': ['Observation', 'Line 1, Col 681'],
                     'severity': 'error'}


case 004:

{'body': {'issue': [{'code': 'processing',
                     'diagnostics': 'The type of element effectiveTiming is not known, which is illegal. Valid types at this point are dateTime, Period',
                     'location': ['Observation', 'Line 1, Col 659'],
                     'severity': 'error'}


case 005:

{'body': {'issue': [{'code': 'processing',
                     'diagnostics': 'This element does not match any known slice defined in the profile https://www.netzwerk-universitaetsmedizin.de/fhir/StructureDefinition/blood-pressure',
                     'location': ['Observation.component[0]',
                                  'Line 1, Col 686'],
                     'severity': 'information'},
                    {'code': 'processing',
                     'diagnostics': 'This element does not match any known slice defined in the profile https://www.netzwerk-universitaetsmedizin.de/fhir/StructureDefinition/blood-pressure',
                     'location': ['Observation.component[1]',
                                  'Line 1, Col 913'],
                     'severity': 'information'},
                    {'code': 'processing',
                     'diagnostics': 'Observation.component:SystolicBP: minimum required = 1, but only found 0 (from https://www.netzwerk-universitaetsmedizin.de/fhir/StructureDefinition/blood-pressure)',
                     'location': ['Observation', 'Line 1, Col 2'],
                     'severity': 'error'},
                    {'code': 'processing',
                     'diagnostics': 'Observation.component:DiastolicBP: minimum required = 1, but only found 0 (from https://www.netzwerk-universitaetsmedizin.de/fhir/StructureDefinition/blood-pressure)',
                     'location': ['Observation', 'Line 1, Col 2'],
                     'severity': 'error'},
                    {'code': 'processing',
                     'diagnostics': 'Observation.code.coding:BPCode: minimum required = 1, but only found 0 (from https://www.netzwerk-universitaetsmedizin.de/fhir/StructureDefinition/blood-pressure)',
                     'location': ['Observation.code', 'Line 1, Col 358'],
                     'severity': 'error'}


case 006:

{'body': {'issue': [{'code': 'processing',
                     'diagnostics': 'This element does not match any known slice defined in the profile https://www.netzwerk-universitaetsmedizin.de/fhir/StructureDefinition/blood-pressure',
                     'location': ['Observation.component[0]',
                                  'Line 1, Col 704'],
                     'severity': 'information'},
                    {'code': 'processing',
                     'diagnostics': 'This element does not match any known slice defined in the profile https://www.netzwerk-universitaetsmedizin.de/fhir/StructureDefinition/blood-pressure',
                     'location': ['Observation.component[1]',
                                  'Line 1, Col 931'],
                     'severity': 'information'},
                    {'code': 'processing',
                     'diagnostics': 'Observation.component:SystolicBP: minimum required = 1, but only found 0 (from https://www.netzwerk-universitaetsmedizin.de/fhir/StructureDefinition/blood-pressure)',
                     'location': ['Observation', 'Line 1, Col 2'],
                     'severity': 'error'},
                    {'code': 'processing',
                     'diagnostics': 'Observation.component:DiastolicBP: minimum required = 1, but only found 0 (from https://www.netzwerk-universitaetsmedizin.de/fhir/StructureDefinition/blood-pressure)',
                     'location': ['Observation', 'Line 1, Col 2'],
                     'severity': 'error'},
                    {'code': 'processing',
                     'diagnostics': 'Observation.code.coding:BPCode: minimum required = 1, but only found 0 (from https://www.netzwerk-universitaetsmedizin.de/fhir/StructureDefinition/blood-pressure)',
                     'location': ['Observation.code', 'Line 1, Col 358'],
                     'severity': 'error'},
                    {'code': 'processing',
                     'diagnostics': 'The type of element effectiveInstant is not known, which is illegal. Valid types at this point are dateTime, Period',
                     'location': ['Observation', 'Line 1, Col 688'],
                     'severity': 'error'},
                    {'code': 'processing',
                     'diagnostics': 'vs-1: if Observation.effective[x] is dateTime and has a value then that value shall be precise to the day [($this as dateTime).toString().length() >= 8]',
                     'location': ['Observation.effective.ofType(instant)',
                                  'Line 1, Col 688'],
                     'severity': 'error'}


case 007:

{'body': {'issue': [{'code': 'processing',
                     'diagnostics': 'This element does not match any known slice defined in the profile https://www.netzwerk-universitaetsmedizin.de/fhir/StructureDefinition/blood-pressure',
                     'location': ['Observation.component[0]',
                                  'Line 1, Col 716'],
                     'severity': 'information'},
                    {'code': 'processing',
                     'diagnostics': 'This element does not match any known slice defined in the profile https://www.netzwerk-universitaetsmedizin.de/fhir/StructureDefinition/blood-pressure',
                     'location': ['Observation.component[1]',
                                  'Line 1, Col 943'],
                     'severity': 'information'},
                    {'code': 'processing',
                     'diagnostics': 'Observation.component:SystolicBP: minimum required = 1, but only found 0 (from https://www.netzwerk-universitaetsmedizin.de/fhir/StructureDefinition/blood-pressure)',
                     'location': ['Observation', 'Line 1, Col 2'],
                     'severity': 'error'},
                    {'code': 'processing',
                     'diagnostics': 'Observation.component:DiastolicBP: minimum required = 1, but only found 0 (from https://www.netzwerk-universitaetsmedizin.de/fhir/StructureDefinition/blood-pressure)',
                     'location': ['Observation', 'Line 1, Col 2'],
                     'severity': 'error'},
                    {'code': 'processing',
                     'diagnostics': 'Observation.code.coding:BPCode: minimum required = 1, but only found 0 (from https://www.netzwerk-universitaetsmedizin.de/fhir/StructureDefinition/blood-pressure)',
                     'location': ['Observation.code', 'Line 1, Col 358'],
                     'severity': 'error'}

case 008:

{'body': {'issue': [{'code': 'processing',
                     'diagnostics': 'This element does not match any known slice defined in the profile https://www.netzwerk-universitaetsmedizin.de/fhir/StructureDefinition/blood-pressure',
                     'location': ['Observation.component[0]',
                                  'Line 1, Col 697'],
                     'severity': 'information'},
                    {'code': 'processing',
                     'diagnostics': 'This element does not match any known slice defined in the profile https://www.netzwerk-universitaetsmedizin.de/fhir/StructureDefinition/blood-pressure',
                     'location': ['Observation.component[1]',
                                  'Line 1, Col 924'],
                     'severity': 'information'},
                    {'code': 'processing',
                     'diagnostics': 'Observation.component:SystolicBP: minimum required = 1, but only found 0 (from https://www.netzwerk-universitaetsmedizin.de/fhir/StructureDefinition/blood-pressure)',
                     'location': ['Observation', 'Line 1, Col 2'],
                     'severity': 'error'},
                    {'code': 'processing',
                     'diagnostics': 'Observation.component:DiastolicBP: minimum required = 1, but only found 0 (from https://www.netzwerk-universitaetsmedizin.de/fhir/StructureDefinition/blood-pressure)',
                     'location': ['Observation', 'Line 1, Col 2'],
                     'severity': 'error'},
                    {'code': 'processing',
                     'diagnostics': 'Observation.code.coding:BPCode: minimum required = 1, but only found 0 (from https://www.netzwerk-universitaetsmedizin.de/fhir/StructureDefinition/blood-pressure)',
                     'location': ['Observation.code', 'Line 1, Col 358'],
                     'severity': 'error'},
                    {'code': 'processing',
                     'diagnostics': 'The type of element effectiveTiming is not known, which is illegal. Valid types at this point are dateTime, Period',
                     'location': ['Observation', 'Line 1, Col 666'],
                     'severity': 'error'},
                    {'code': 'processing',
                     'diagnostics': 'vs-1: if Observation.effective[x] is dateTime and has a value then that value shall be precise to the day [($this as dateTime).toString().length() >= 8]',
                     'location': ['Observation.effective.ofType(Timing)',
                                  'Line 1, Col 666'],
                     'severity': 'error'}



## Expected Result

200



STACKTRACE ONLY FOR 004:

2021-02-08 17:53:12.615 ERROR 11374 --- [io-8888-exec-10] o.a.c.p.e.DefaultErrorHandler            : Failed delivery for (MessageId: D1C812AAE75FB97-00000000000000FB on ExchangeId: D1C812AAE75FB97-00000000000000FB). Exhausted after delivery attempt: 1 caught: org.hl7.fhir.exceptions.FHIRException: Type mismatch: the type DateTimeType was expected, but org.hl7.fhir.r4.model.Period was encountered. Processed by failure processor: FatalFallbackErrorHandler[Channel[DelegateSync[org.ehrbase.fhirbridge.camel.processor.DefaultExceptionHandler@5d4af473]]]

Message History (complete message history is disabled)
---------------------------------------------------------------------------------------------------------------------------------------
RouteId              ProcessorId          Processor                                                                        Elapsed (ms)
[route5            ] [route5            ] [from[fhir-create-observation://fhirConsumer?fhirContext=%23fhirContext]       ] [       481]

Stacktrace
---------------------------------------------------------------------------------------------------------------------------------------


org.hl7.fhir.exceptions.FHIRException: Type mismatch: the type DateTimeType was expected, but org.hl7.fhir.r4.model.Period was encountered
	at org.hl7.fhir.r4.model.Observation.getEffectiveDateTimeType(Observation.java:2024) ~[org.hl7.fhir.r4-5.2.0.jar!/:na]
	at org.ehrbase.fhirbridge.ehr.converter.BloodPressureCompositionConverter.toComposition(BloodPressureCompositionConverter.java:43) ~[classes!/:na]
	at org.ehrbase.fhirbridge.ehr.converter.BloodPressureCompositionConverter.toComposition(BloodPressureCompositionConverter.java:18) ~[classes!/:na]
	at org.ehrbase.fhirbridge.camel.component.ehr.composition.CompositionProducer.mergeCompositionEntity(CompositionProducer.java:43) ~[classes!/:na]
	at org.ehrbase.fhirbridge.camel.component.ehr.composition.CompositionProducer.process(CompositionProducer.java:27) ~[classes!/:na]
	at org.apache.camel.support.AsyncProcessorConverterHelper$ProcessorToAsyncProcessorBridge.process(AsyncProcessorConverterHelper.java:66) ~[camel-support-3.7.0.jar!/:3.7.0]
	at org.apache.camel.processor.SendProcessor.process(SendProcessor.java:169) ~[camel-core-processor-3.7.0.jar!/:3.7.0]
	at org.apache.camel.processor.errorhandler.RedeliveryErrorHandler$RedeliveryTask.doRun(RedeliveryErrorHandler.java:714) ~[camel-core-processor-3.7.0.jar!/:3.7.0]
	at org.apache.camel.processor.errorhandler.RedeliveryErrorHandler$RedeliveryTask.run(RedeliveryErrorHandler.java:623) ~[camel-core-processor-3.7.0.jar!/:3.7.0]
	at org.apache.camel.impl.engine.DefaultReactiveExecutor$Worker.schedule(DefaultReactiveExecutor.java:148) ~[camel-base-engine-3.7.0.jar!/:3.7.0]
	at org.apache.camel.impl.engine.DefaultReactiveExecutor.scheduleMain(DefaultReactiveExecutor.java:60) ~[camel-base-engine-3.7.0.jar!/:3.7.0]
	at org.apache.camel.processor.Pipeline.process(Pipeline.java:147) ~[camel-core-processor-3.7.0.jar!/:3.7.0]
	at org.apache.camel.impl.engine.CamelInternalProcessor.process(CamelInternalProcessor.java:312) ~[camel-base-engine-3.7.0.jar!/:3.7.0]
	at org.apache.camel.impl.engine.DefaultAsyncProcessorAwaitManager.process(DefaultAsyncProcessorAwaitManager.java:83) ~[camel-base-engine-3.7.0.jar!/:3.7.0]
	at org.apache.camel.support.AsyncProcessorSupport.process(AsyncProcessorSupport.java:41) ~[camel-support-3.7.0.jar!/:3.7.0]
	at org.openehealth.ipf.platform.camel.ihe.fhir.core.intercept.consumer.ConsumerAuditInterceptor.process(ConsumerAuditInterceptor.java:60) ~[ipf-platform-camel-ihe-fhir-core-4.0.0.jar!/:na]
	at org.openehealth.ipf.platform.camel.ihe.fhir.core.FhirConsumer.runRoute(FhirConsumer.java:192) ~[ipf-platform-camel-ihe-fhir-core-4.0.0.jar!/:na]
	at org.openehealth.ipf.platform.camel.ihe.fhir.core.FhirConsumer.handleInRoute(FhirConsumer.java:165) ~[ipf-platform-camel-ihe-fhir-core-4.0.0.jar!/:na]
	at org.openehealth.ipf.platform.camel.ihe.fhir.core.FhirConsumer.handleAction(FhirConsumer.java:137) ~[ipf-platform-camel-ihe-fhir-core-4.0.0.jar!/:na]
	at org.openehealth.ipf.commons.ihe.fhir.AbstractPlainProvider.requestAction(AbstractPlainProvider.java:170) ~[ipf-commons-ihe-fhir-core-4.0.0.jar!/:na]
	at org.ehrbase.fhirbridge.fhir.observation.CreateObservationProvider.create(CreateObservationProvider.java:20) ~[classes!/:na]
	at jdk.internal.reflect.GeneratedMethodAccessor614.invoke(Unknown Source) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:564) ~[na:na]
	at ca.uhn.fhir.rest.server.method.BaseMethodBinding.invokeServerMethod(BaseMethodBinding.java:245) ~[hapi-fhir-server-5.2.0.jar!/:na]
	at ca.uhn.fhir.rest.server.method.BaseOutcomeReturningMethodBinding.invokeServer(BaseOutcomeReturningMethodBinding.java:154) ~[hapi-fhir-server-5.2.0.jar!/:na]
	at ca.uhn.fhir.rest.server.method.CreateMethodBinding.invokeServer(CreateMethodBinding.java:41) ~[hapi-fhir-server-5.2.0.jar!/:na]
	at ca.uhn.fhir.rest.server.RestfulServer.handleRequest(RestfulServer.java:1052) ~[hapi-fhir-server-5.2.0.jar!/:na]
	at ca.uhn.fhir.rest.server.RestfulServer.doPost(RestfulServer.java:377) ~[hapi-fhir-server-5.2.0.jar!/:na]
	at ca.uhn.fhir.rest.server.RestfulServer.service(RestfulServer.java:1749) ~[hapi-fhir-server-5.2.0.jar!/:na]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:733) ~[tomcat-embed-core-9.0.41.jar!/:4.0.FR]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53) ~[tomcat-embed-websocket-9.0.41.jar!/:9.0.41]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.springframework.web.filter.CorsFilter.doFilterInternal(CorsFilter.java:91) ~[spring-web-5.3.2.jar!/:5.3.2]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.3.2.jar!/:5.3.2]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100) ~[spring-web-5.3.2.jar!/:5.3.2]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.3.2.jar!/:5.3.2]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93) ~[spring-web-5.3.2.jar!/:5.3.2]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.3.2.jar!/:5.3.2]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.springframework.boot.actuate.metrics.web.servlet.WebMvcMetricsFilter.doFilterInternal(WebMvcMetricsFilter.java:93) ~[spring-boot-actuator-2.4.1.jar!/:2.4.1]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.3.2.jar!/:5.3.2]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201) ~[spring-web-5.3.2.jar!/:5.3.2]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.3.2.jar!/:5.3.2]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:542) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:143) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:374) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:888) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1597) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) ~[na:na]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at java.base/java.lang.Thread.run(Thread.java:832) ~[na:na]

2021-02-08 17:53:12.721 ERROR 11374 --- [io-8888-exec-10] c.u.f.r.s.i.ExceptionHandlingInterceptor : Failure during REST processing

ca.uhn.fhir.rest.server.exceptions.InternalErrorException: Unexpected server error
	at org.openehealth.ipf.platform.camel.ihe.fhir.core.FhirConsumer.runRoute(FhirConsumer.java:203) ~[ipf-platform-camel-ihe-fhir-core-4.0.0.jar!/:na]
	at org.openehealth.ipf.platform.camel.ihe.fhir.core.FhirConsumer.handleInRoute(FhirConsumer.java:165) ~[ipf-platform-camel-ihe-fhir-core-4.0.0.jar!/:na]
	at org.openehealth.ipf.platform.camel.ihe.fhir.core.FhirConsumer.handleAction(FhirConsumer.java:137) ~[ipf-platform-camel-ihe-fhir-core-4.0.0.jar!/:na]
	at org.openehealth.ipf.commons.ihe.fhir.AbstractPlainProvider.requestAction(AbstractPlainProvider.java:170) ~[ipf-commons-ihe-fhir-core-4.0.0.jar!/:na]
	at org.ehrbase.fhirbridge.fhir.observation.CreateObservationProvider.create(CreateObservationProvider.java:20) ~[classes!/:na]
	at jdk.internal.reflect.GeneratedMethodAccessor614.invoke(Unknown Source) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:564) ~[na:na]
	at ca.uhn.fhir.rest.server.method.BaseMethodBinding.invokeServerMethod(BaseMethodBinding.java:245) ~[hapi-fhir-server-5.2.0.jar!/:na]
	at ca.uhn.fhir.rest.server.method.BaseOutcomeReturningMethodBinding.invokeServer(BaseOutcomeReturningMethodBinding.java:154) ~[hapi-fhir-server-5.2.0.jar!/:na]
	at ca.uhn.fhir.rest.server.method.CreateMethodBinding.invokeServer(CreateMethodBinding.java:41) ~[hapi-fhir-server-5.2.0.jar!/:na]
	at ca.uhn.fhir.rest.server.RestfulServer.handleRequest(RestfulServer.java:1052) ~[hapi-fhir-server-5.2.0.jar!/:na]
	at ca.uhn.fhir.rest.server.RestfulServer.doPost(RestfulServer.java:377) ~[hapi-fhir-server-5.2.0.jar!/:na]
	at ca.uhn.fhir.rest.server.RestfulServer.service(RestfulServer.java:1749) ~[hapi-fhir-server-5.2.0.jar!/:na]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:733) ~[tomcat-embed-core-9.0.41.jar!/:4.0.FR]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53) ~[tomcat-embed-websocket-9.0.41.jar!/:9.0.41]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.springframework.web.filter.CorsFilter.doFilterInternal(CorsFilter.java:91) ~[spring-web-5.3.2.jar!/:5.3.2]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.3.2.jar!/:5.3.2]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100) ~[spring-web-5.3.2.jar!/:5.3.2]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.3.2.jar!/:5.3.2]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93) ~[spring-web-5.3.2.jar!/:5.3.2]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.3.2.jar!/:5.3.2]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.springframework.boot.actuate.metrics.web.servlet.WebMvcMetricsFilter.doFilterInternal(WebMvcMetricsFilter.java:93) ~[spring-boot-actuator-2.4.1.jar!/:2.4.1]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.3.2.jar!/:5.3.2]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201) ~[spring-web-5.3.2.jar!/:5.3.2]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.3.2.jar!/:5.3.2]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:542) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:143) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:374) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:888) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1597) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) ~[na:na]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) ~[tomcat-embed-core-9.0.41.jar!/:9.0.41]
	at java.base/java.lang.Thread.run(Thread.java:832) ~[na:na]
Caused by: org.hl7.fhir.exceptions.FHIRException: Type mismatch: the type DateTimeType was expected, but org.hl7.fhir.r4.model.Period was encountered
	at org.hl7.fhir.r4.model.Observation.getEffectiveDateTimeType(Observation.java:2024) ~[org.hl7.fhir.r4-5.2.0.jar!/:na]
	at org.ehrbase.fhirbridge.ehr.converter.BloodPressureCompositionConverter.toComposition(BloodPressureCompositionConverter.java:43) ~[classes!/:na]
	at org.ehrbase.fhirbridge.ehr.converter.BloodPressureCompositionConverter.toComposition(BloodPressureCompositionConverter.java:18) ~[classes!/:na]
	at org.ehrbase.fhirbridge.camel.component.ehr.composition.CompositionProducer.mergeCompositionEntity(CompositionProducer.java:43) ~[classes!/:na]
	at org.ehrbase.fhirbridge.camel.component.ehr.composition.CompositionProducer.process(CompositionProducer.java:27) ~[classes!/:na]
	at org.apache.camel.support.AsyncProcessorConverterHelper$ProcessorToAsyncProcessorBridge.process(AsyncProcessorConverterHelper.java:66) ~[camel-support-3.7.0.jar!/:3.7.0]
	at org.apache.camel.processor.SendProcessor.process(SendProcessor.java:169) ~[camel-core-processor-3.7.0.jar!/:3.7.0]
	at org.apache.camel.processor.errorhandler.RedeliveryErrorHandler$RedeliveryTask.doRun(RedeliveryErrorHandler.java:714) ~[camel-core-processor-3.7.0.jar!/:3.7.0]
	at org.apache.camel.processor.errorhandler.RedeliveryErrorHandler$RedeliveryTask.run(RedeliveryErrorHandler.java:623) ~[camel-core-processor-3.7.0.jar!/:3.7.0]
	at org.apache.camel.impl.engine.DefaultReactiveExecutor$Worker.schedule(DefaultReactiveExecutor.java:148) ~[camel-base-engine-3.7.0.jar!/:3.7.0]
	at org.apache.camel.impl.engine.DefaultReactiveExecutor.scheduleMain(DefaultReactiveExecutor.java:60) ~[camel-base-engine-3.7.0.jar!/:3.7.0]
	at org.apache.camel.processor.Pipeline.process(Pipeline.java:147) ~[camel-core-processor-3.7.0.jar!/:3.7.0]
	at org.apache.camel.impl.engine.CamelInternalProcessor.process(CamelInternalProcessor.java:312) ~[camel-base-engine-3.7.0.jar!/:3.7.0]
	at org.apache.camel.impl.engine.DefaultAsyncProcessorAwaitManager.process(DefaultAsyncProcessorAwaitManager.java:83) ~[camel-base-engine-3.7.0.jar!/:3.7.0]
	at org.apache.camel.support.AsyncProcessorSupport.process(AsyncProcessorSupport.java:41) ~[camel-support-3.7.0.jar!/:3.7.0]
	at org.openehealth.ipf.platform.camel.ihe.fhir.core.intercept.consumer.ConsumerAuditInterceptor.process(ConsumerAuditInterceptor.java:60) ~[ipf-platform-camel-ihe-fhir-core-4.0.0.jar!/:na]
	at org.openehealth.ipf.platform.camel.ihe.fhir.core.FhirConsumer.runRoute(FhirConsumer.java:192) ~[ipf-platform-camel-ihe-fhir-core-4.0.0.jar!/:na]
	... 55 common frames omitted

--- a/src/main/java/org/ehrbase/fhirbridge/ehr/converter/specific/patientinicu/PatientInIcuCompositionConverter.java
+++ b/src/main/java/org/ehrbase/fhirbridge/ehr/converter/specific/patientinicu/PatientInIcuCompositionConverter.java
@@ -7,8 +7,6 @@ import org.ehrbase.fhirbridge.ehr.opt.patientauficucomposition.definition.Status
 import org.hl7.fhir.r4.model.Observation;
 import org.springframework.lang.NonNull;
 
-import java.util.HashMap;
-
 public class PatientInIcuCompositionConverter extends CompositionConverter<Observation, PatientAufICUComposition> {
 
     private  void setStatus(PatientAufICUComposition composition, Observation fhirObservation) {

--- a/src/test/resources/Observation/create-blood-pressure_snomed-datetime.json
+++ b/src/test/resources/Observation/create-blood-pressure_snomed-datetime.json
@@ -21,6 +21,10 @@
   "code": {
     "coding": [
       {
+        "system": "http://loinc.org",
+        "code": "85354-9"
+      },
+      {
         "system": "http://snomed.info/sct",
         "code": "75367002",
         "display": "Blood pressure panel with all children optional"
@@ -40,6 +44,10 @@
       "code": {
         "coding": [
           {
+            "system": "http://loinc.org",
+            "code": "8480-6"
+          },
+          {
             "system": "http://snomed.info/sct",
             "code": "271649006",
             "display": "Systolic blood pressure"
@@ -56,6 +64,10 @@
     {
       "code": {
         "coding": [
+          {
+            "system": "http://loinc.org",
+            "code": "8462-4"
+          },
           {
             "system": "http://snomed.info/sct",
             "code": "271650006",

--- a/src/test/resources/Observation/create-blood-pressure_snomed-instant.json
+++ b/src/test/resources/Observation/create-blood-pressure_snomed-instant.json
@@ -21,6 +21,10 @@
   "code": {
     "coding": [
       {
+        "system": "http://loinc.org",
+        "code": "85354-9"
+      },
+      {
         "system": "http://snomed.info/sct",
         "code": "75367002",
         "display": "Blood pressure panel with all children optional"
@@ -40,6 +44,10 @@
       "code": {
         "coding": [
           {
+            "system": "http://loinc.org",
+            "code": "8480-6"
+          },
+          {
             "system": "http://snomed.info/sct",
             "code": "271649006",
             "display": "Systolic blood pressure"
@@ -56,6 +64,10 @@
     {
       "code": {
         "coding": [
+          {
+            "system": "http://loinc.org",
+            "code": "8462-4"
+          },
           {
             "system": "http://snomed.info/sct",
             "code": "271650006",

--- a/src/test/resources/Observation/create-blood-pressure_snomed-period.json
+++ b/src/test/resources/Observation/create-blood-pressure_snomed-period.json
@@ -21,6 +21,10 @@
   "code": {
     "coding": [
       {
+        "system": "http://loinc.org",
+        "code": "85354-9"
+      },
+      {
         "system": "http://snomed.info/sct",
         "code": "75367002",
         "display": "Blood pressure panel with all children optional"
@@ -43,6 +47,10 @@
       "code": {
         "coding": [
           {
+            "system": "http://loinc.org",
+            "code": "8480-6"
+          },
+          {
             "system": "http://snomed.info/sct",
             "code": "271649006",
             "display": "Systolic blood pressure"
@@ -59,6 +67,10 @@
     {
       "code": {
         "coding": [
+          {
+            "system": "http://loinc.org",
+            "code": "8462-4"
+          },
           {
             "system": "http://snomed.info/sct",
             "code": "271650006",

--- a/src/test/resources/Observation/create-blood-pressure_snomed-timing.json
+++ b/src/test/resources/Observation/create-blood-pressure_snomed-timing.json
@@ -21,6 +21,10 @@
   "code": {
     "coding": [
       {
+        "system": "http://loinc.org",
+        "code": "85354-9"
+      },
+      {
         "system": "http://snomed.info/sct",
         "code": "75367002",
         "display": "Blood pressure panel with all children optional"
@@ -44,6 +48,10 @@
       "code": {
         "coding": [
           {
+            "system": "http://loinc.org",
+            "code": "8480-6"
+          },
+          {
             "system": "http://snomed.info/sct",
             "code": "271649006",
             "display": "Systolic blood pressure"
@@ -60,6 +68,10 @@
     {
       "code": {
         "coding": [
+          {
+            "system": "http://loinc.org",
+            "code": "8462-4"
+          },
           {
             "system": "http://snomed.info/sct",
             "code": "271650006",

--- a/tests/robot/OBSERVATION/01_create.robot
+++ b/tests/robot/OBSERVATION/01_create.robot
@@ -820,11 +820,11 @@ Force Tags              observation_create    create
 	...                 3. *UPDATE* ``Subject - Identifier - value`` with the _UUID:_ ${subject_id} which was created in EHR record\n\n
     ...                 4. *POST* example JSON to observation endpoint\n\n
 	...                 5. *VALIDATE* the response status
-	[Tags]             	blood-pressure    valid    alternative   loinc    instant    not-ready_bug    not-ready    194
+	[Tags]             	blood-pressure    invalid    instant not-ready_bug
 
     ehr.create new ehr    000_ehr_status.json
     observation.create blood pressure  Blood Pressure  create-blood-pressure_loinc-instant.json
-    observation.validate response - 201
+    observation.validate response - 422 (with error message)        422         The type of element effectiveInstant is not known, which is illegal
 
 
 
@@ -848,11 +848,11 @@ Force Tags              observation_create    create
 	...                 3. *UPDATE* ``Subject - Identifier - value`` with the _UUID:_ ${subject_id} which was created in EHR record\n\n
     ...                 4. *POST* example JSON to observation endpoint\n\n
 	...                 5. *VALIDATE* the response status
-	[Tags]             	blood-pressure    valid    alternative    loinc    timing    not-ready_bug    not-ready    194
+	[Tags]             	blood-pressure    invalid    timing
 
     ehr.create new ehr    000_ehr_status.json
     observation.create blood pressure  Blood Pressure  create-blood-pressure_loinc-timing.json
-    observation.validate response - 201
+    observation.validate response - 422 (with error message)        422         The type of element effectiveTiming is not known, which is illegal
 
 
 
@@ -862,7 +862,7 @@ Force Tags              observation_create    create
 	...                 3. *UPDATE* ``Subject - Identifier - value`` with the _UUID:_ ${subject_id} which was created in EHR record\n\n
     ...                 4. *POST* example JSON to observation endpoint\n\n
 	...                 5. *VALIDATE* the response status
-	[Tags]             	blood-pressure    valid    alternative    snomed    datetime    not-ready_bug    not-ready    194
+	[Tags]             	blood-pressure    valid    alternative    snomed    datetime
 
     ehr.create new ehr    000_ehr_status.json
     observation.create blood pressure  Blood Pressure  create-blood-pressure_snomed-datetime.json
@@ -876,11 +876,11 @@ Force Tags              observation_create    create
 	...                 3. *UPDATE* ``Subject - Identifier - value`` with the _UUID:_ ${subject_id} which was created in EHR record\n\n
     ...                 4. *POST* example JSON to observation endpoint\n\n
 	...                 5. *VALIDATE* the response status
-	[Tags]             	blood-pressure    valid    alternative    snomed    instant    not-ready_bug    not-ready    194
+	[Tags]             	blood-pressure    invalid    instant
 
     ehr.create new ehr    000_ehr_status.json
     observation.create blood pressure  Blood Pressure  create-blood-pressure_snomed-instant.json
-    observation.validate response - 201
+    observation.validate response - 422 (with error message)        422         The type of element effectiveInstant is not known, which is illegal
 
 
 
@@ -890,7 +890,7 @@ Force Tags              observation_create    create
 	...                 3. *UPDATE* ``Subject - Identifier - value`` with the _UUID:_ ${subject_id} which was created in EHR record\n\n
     ...                 4. *POST* example JSON to observation endpoint\n\n
 	...                 5. *VALIDATE* the response status
-	[Tags]             	blood-pressure    valid    alternative    snomed    period    not-ready_bug    not-ready    194
+	[Tags]             	blood-pressure    valid    alternative    snomed    period
 
     ehr.create new ehr    000_ehr_status.json
     observation.create blood pressure  Blood Pressure  create-blood-pressure_snomed-period.json
@@ -904,11 +904,11 @@ Force Tags              observation_create    create
 	...                 3. *UPDATE* ``Subject - Identifier - value`` with the _UUID:_ ${subject_id} which was created in EHR record\n\n
     ...                 4. *POST* example JSON to observation endpoint\n\n
 	...                 5. *VALIDATE* the response status
-	[Tags]             	blood-pressure    valid    alternative    snomed    timing    not-ready_bug    not-ready    194
+	[Tags]             	blood-pressure    invalid    timing
 
     ehr.create new ehr    000_ehr_status.json
     observation.create blood pressure  Blood Pressure  create-blood-pressure_snomed-timing.json
-    observation.validate response - 201
+    observation.validate response - 422 (with error message)        422         The type of element effectiveTiming is not known, which is illegal
 
 
 

